op-supervisor: Write logs to logdb

ajsutton opened the pull request:
**Description**

Implements the hash scheme for logs and actually writes them to the logdb.

Executing messages aren't extracted yet but basic log information is hashed and written (tracked in #11024). Also doesn't restore state properly on startup, always starting from block 0 (tracked in #11027)

**Tests**

Added unit tests. Confirmed logs are written when running locally.

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes #11025

protolambda commented on op-supervisor/supervisor/backend/source/log_processor.go:
```diff
@@ -0,0 +1,55 @@
+	return db.TruncateHash(crypto.Keccak256Hash(msg))
+}
+
+func logToPayload(l *types.Log) []byte {
+	msg := make([]byte, 0)
+	for _, topic := range l.Topics {
+		msg = append(msg, topic.Bytes()...)
+	}
+	msg = append(msg, l.Data...)
+	return msg
+}
```
Should we encode the number of topics as prefix, so we know where the trailing non-topic data starts?

ajsutton replied:
I was following https://specs.optimism.io/interop/messaging.html#message-payload

We could change that if there's a reason to, but would need to update the contracts as well so I think they're just emitting the payload hash now rather than the full payload data.  In terms of our use cases I don't think we ever decode that message payload - we just want to check it matches what was claimed by the executing message.

For other people using this, if you have the original ABI then you would know how many topics there are and would have the other info like types required to decode the whole message. According to the spec you can just use `abi.decode` though I don't know enough about `abi.decode` to know that for sure.

ajsutton replied:
ah the change to only emit the payload hash is still a spec PR (https://github.com/ethereum-optimism/specs/pull/254/).

protolambda reviewed (commented):
Looks good, but I am unsure about the log-payload hashing. I think we need to specify better what properies we include in the hash, and how it is constructed, and maybe include some versioning.

ajsutton committed:
op-supervisor: Write logs to logdb

Executing messages aren't extracted yet but basic log information is hashed and written.
